lms/djangoapps/courseware/views/views.py

```diff
@@ -1662,7 +1675,31 @@ def render_public_video_xblock(request, usage_key_string):
             'is_learning_mfe': True,
             'is_mobile_app': False,
         }
-        return render_to_response('courseware/courseware-chromeless.html', context)
+        return render_to_response(template, context)
+
+
+@require_http_methods(["GET"])
+@ensure_valid_usage_key
+@xframe_options_exempt
+@transaction.non_atomic_requests
+def render_public_video_xblock_embed(request, usage_key_string):
+    """
+    Returns an HttpResponse with HTML content for the Video xBlock with the given usage_key.
+    The returned HTML consists of nothing but the Video xBlock content for use in social media embedding.
+    """
+    return _render_public_video_xblock(request, usage_key_string, is_=True)
+
+
+@require_http_methods(["GET"])
+@ensure_valid_usage_key
+@xframe_options_exempt
+@transaction.non_atomic_requests
+def render_public_video_xblock(request, usage_key_string):
+    """
+    Returns an HttpResponse with HTML content for the Video xBlock with the given usage_key.
+    The returned HTML is a chromeless rendering of the Video xBlock (excluding content of the containing courseware).
+    """
+    return _render_public_video_xblock(request, usage_key_string, is_=False)
```

Comment:
I also don't love the pattern of wrapping private functions where the private function has *nearly all* of the logic. This suggests that we no longer want to test anything inside of those private functions.

Reply:
I don't agree with that last comment. We can and should test private functions.